python/simulation/fake_raft.py

```diff
@@ -0,0 +1,509 @@
+    def slot_names(self):
+        """ The names of the 'slots' associated with the sensors """
+        sl = self.__sensor_dict.keys()
+        sl.sort()
+        return sl
+
+
+    @property
+    def sensor_names(self):
+        """ The names of the sensors in this raft, sorted to match the slot names """
+        sl = self.slot_names
+        rl = [ self.__sensor_dict[s].sensor_id for s in sl]
+        return rl
+
+    def items(self):
+        """ Iterator over slot_name, sensor_name pairs """
+        sl = self.slot_names
+        sn = self.sensor_names
+        return zip(sl,sn)
+
+    def sensor(self,slot):
+        """ Sensor associated with a particular slot """
+        return self.__sensor_dict[slot]
+
+    def file_copy(self,process_name,
+                  output_path,pattern='*.fits',
+                  clobber=True,dry_run=False):
+        """ Copy a single input file to the correct output location for each sensor in this raft.
+            Possibly updating FITS header infomation along the way.
+
+        Parameters
+        ----------
+        process_name : str
+                    The name of the assoicated eTraveler process, used in making the output file name
+              
+        output_path : str
+                    The prefix for the output file paths        
+
+        pattern : str
+                    Regular expression specifying which files to get
+
+        clobber : bool, optional
+                  Allow overwriting existing files
+
+        dry_run : bool, optional
+            If true, just print output file names, but do not copy files
+         """        
+        writer = RaftImages(self.__raft_id,process_name,self.sensor_type,output_path)
+
+        for k,v in self.items():
+            template_files = self.get_template_files(process_name,pattern,slot=k)
+            for f in template_files:
+                writer.write_sensor_image(f,v,clobber=clobber,dry_run=dry_run)
+            
+            
+    def get_template_files(self,process_name,pattern='*.fits',slot=None):        
+        """ Get examples of the input files associated to a particular process.
+
+        Parameters
+        ----------
+        process_name : str
+                    The name of the assoicated eTraveler process, used in making the output file name
+              
+        pattern : str
+                    Regular expression specifying which files to get
+ 
+        slot : str
+                   If set, use the sensor associated with this slot as the template
+
+        Returns
+        ----------
+        file_list : list
+
+        List of file names for files that match the process_name, sensor_id and pattern
+        """        
+        if slot is None:
+            sn = self.sensor_names[0]
+        else:
+            sn = self.sensor(slot).sensor_id
+
+        return get_template_files(sn,process_name,pattern)
+
+
+
+if __name__ == '__main__':
+
+    outpath = 'output/'
+    process_name = 'fe55_offline'
```

Comment:
If we use TS3 data here, the `root_folder` should be `/LSST/mirror/BNL-prod/prod` and `process_name`s like `fe55_acq` would be appropriate, but we may also want to use vendor data, in which case to retrieve the data, the process name `vendorIngest` would be used, but the `process_name` on output should be different, e.g., 'fe55_acq', etc..  Also note that the vendor data are all in the same folder for the different image and test types.  So I think querying by `TESTTYPE` and `IMGTYPE` metadata keywords would be desirable, though file pattern globbing would work.  OTOH, we may decide that we never want to use vendor data for these simulations, in which case, what you have now will work, but it may be worth thinking about what it would take to make it possible to use vendor data.